manifests/init.pp

```diff
@@ -50,22 +50,30 @@
 #  If set to `['never']`, then this module will not create any incidents at all.
 class servicenow_reporting_integration (
   String[1] $instance,
-  String[1] $caller_id,
-  Optional[String[1]] $pe_console_url                = undef,
-  String $servicenow_credentials_validation_table    = 'incident',
-  Optional[String[1]] $user                          = undef,
-  Optional[String[1]] $password                      = undef,
-  Optional[String[1]] $oauth_token                   = undef,
-  Optional[String[1]] $category                      = undef,
-  Optional[String[1]] $subcategory                   = undef,
-  Optional[String[1]] $contact_type                  = undef,
-  Optional[Integer] $state                           = undef,
-  Optional[Integer] $impact                          = undef,
-  Optional[Integer] $urgency                         = undef,
-  Optional[String[1]] $assignment_group              = undef,
-  Optional[String[1]] $assigned_to                   = undef,
+  Optional[String[1]] $user                                                                  = undef,
+  Optional[String[1]] $password                                                              = undef,
+  Optional[String[1]] $oauth_token                                                           = undef,
+  Enum['event_management', 'incident_management'] $operation_mode                            = 'event_management',
```

Comment:
I felt like this made more sense as a parameter name instead of a boolean like `use_event_management` or w/e. Makes some of the code clearer (e.g. like the caller_id check below).

I will be honest in saying that I can't imagine other modes we could have here so if people are strongly against this, I am happy to use a Boolean parameter.

Also, I was thinking it'd make sense to default to `event_management` since it seems like something the majority of ServiceNow customers would have. Would be nice to double check that though.

Reply:
I can imagine a world where both is a thing. Create incidents for immediate action, but also send events for long term analysis maybe. Either way I like the Enum.